src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -47,7 +47,14 @@ private State convertState(String type) {
 
     private Transport getTransport() throws IOException, JadbException {
         Transport transport = transportFactory.createTransport();
-        send(transport, serial == null ? "host:transport-any" : "host:transport:" + serial );
+        // Do not use try-with-resources here. We want to return unclosed Transport and it is up to caller
+        // to close it. Here we close it only in case of exception.
+        try {
+            send(transport, serial == null ? "host:transport-any" : "host:transport:" + serial );
+        } catch (IOException|JadbException e) {
+            transport.close();
+            throw e;
+        }
```

Comment:
will it not be closed by caller anyway? 🤔

Reply:
Transport can't be closed by caller - the caller will receive exception, not the transport. Caller can close the underlying socket, that will in turn close the transport.